File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.py
import heapq
from collections import defaultdict

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        heap = []
        hash_map = defaultdict(int)

        for num in nums:
            hash_map[num] += 1
        
        for key, val in hash_map.items():
            heapq.heappush(heap, (val, key))
            if len(heap) > k:
                heapq.heappop(heap)

        res = []
        for i in range(k):
            res.append(heapq.heappop(heap)[1])
        return res

        

File: Data Structures & Algorithms/concatenation-of-array/submission-3.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        vector<int> ans;
        int n = nums.size();
        for (int i = 0; i < n * 2; i++) {
            ans.push_back(nums[i % n]);
        }

        return ans;
    }
};